charts/connectivity-exporter/rules/error-budget.yaml

```diff
@@ -128,6 +132,82 @@ groups:
           (service:sli_seconds:total{year!="", month!=""})
       ))
 
+  - record: service:sli_seconds:moving:total
+    expr: |
+      service:sli_seconds
+      +
+      (
+        max_over_time(
+          service:sli_seconds:moving:total[10m]
+        )
+        or
+        service:sli_seconds * 0
+      )
+
+  - record: downtime
+    labels:
+      kind: moving
+      in: hour
+    expr: |
+      (
+        service:sli_seconds:moving:total
+        -
+        (
+          service:sli_seconds:moving:total offset 1h
+          or
+          service:sli_seconds:moving:total * 0
+        )
+      )
+      / 60 / 60
+
+  - record: downtime
+    labels:
+      kind: moving
+      in: day
+    expr: |
+      (
+        service:sli_seconds:moving:total
+        -
+        (
+          service:sli_seconds:moving:total offset 1d
+          or
+          service:sli_seconds:moving:total * 0
+        )
+      )
+      / 60 / 60 / 24
+
+  - record: downtime
+    labels:
+      kind: moving
+      in: week
+    expr: |
+      (
+        service:sli_seconds:moving:total
+        -
+        (
+          service:sli_seconds:moving:total offset 1w
+          or
+          service:sli_seconds:moving:total * 0
+        )
+      )
+      / 60 / 60 / 24 / 7
+
+  - record: downtime
+    labels:
+      kind: moving
+      in: four_weeks
```

Comment:
One open question is if we should do 4 weeks or 30 days for the moving window "month". 4 weeks is nice because it builds on the previous recording rule, but 30 days is usually closer to the length of a month.

Reply:
I think I chose 4 weeks because I remember reading something about it in the SRE book. I looked it up now:

https://sre.google/workbook/implementing-slos/
> We recommend defining this period as an integral number of weeks so it always contains the same number of weekends. For example, if you use a 30-day window, some periods might include four weekends while others include five weekends. If weekend traffic differs significantly from weekday traffic, your SLIs may vary for uninteresting reasons.